### PR TITLE
add encoder and tests

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala [hunk 1]
```diff
@@ -70,9 +74,12 @@ object WebApiRoutes {
         handleRequest[Unit, B](req, _ => f)
     }
 
-    // TODO: Create generic encoders/decoders for
-    // ADT's with discriminator field
+    implicit val encodeBongInfo: Encoder[BondInfo] = deriveEncoder[BondInfo]
```
Comment (tgrospic) [inline node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala]: I can compile it without this line.

File: node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala [hunk 2]
```diff
@@ -70,9 +74,12 @@ object WebApiRoutes {
...
+    implicit val encodeByteString: Encoder[ByteString] =
+      Encoder.encodeString.contramap[ByteString](b => PrettyPrinter.buildStringNoLimit(b))
```
Comment (tgrospic) [inline node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala]: You can supply function directly without creating another lambda.

Comment (zsluedem) [reply node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala]: Do you mean
```
Encoder.encodeString.contramap[ByteString](PrettyPrinter.buildStringNoLimit)
```
?

File: node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala [hunk 3]
```diff
@@ -70,9 +74,12 @@ object WebApiRoutes {
         handleRequest[Unit, B](req, _ => f)
     }
 
-    // TODO: Create generic encoders/decoders for
-    // ADT's with discriminator field
```
Comment (tgrospic) [inline node/src/main/scala/coop/rchain/node/web/WebApiRoutes.scala:74]: Please leave these comments. JSON with discriminator field for case classes is I think better for clients to work with.

**Pull request merged**